File: src/main/java/no/novari/flyt/configuration/validation/constraints/AtLeastOneChildValidator.java
package no.novari.flyt.configuration.validation.constraints;

import no.novari.flyt.configuration.model.configuration.dtos.ObjectMappingDto;
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext;

public class AtLeastOneChildValidator implements HibernateConstraintValidator<AtLeastOneChild, ObjectMappingDto> {

    @Override
    public boolean isValid(ObjectMappingDto value, HibernateConstraintValidatorContext hibernateConstraintValidatorContext) {
        return !value.getValueMappingPerKey().isEmpty()
                || !value.getValueCollectionMappingPerKey().isEmpty()
                || !value.getObjectMappingPerKey().isEmpty()
                || !value.getObjectCollectionMappingPerKey().isEmpty();
    }

}

File: src/main/kotlin/no/novari/flyt/configuration/validation/constraints/AtLeastOneChildValidator.kt
package no.novari.flyt.configuration.validation.constraints

import no.novari.flyt.configuration.model.configuration.dtos.ObjectMappingDto
import org.hibernate.validator.constraintvalidation.HibernateConstraintValidatorContext

class AtLeastOneChildValidator : HibernateConstraintValidator<AtLeastOneChild, ObjectMappingDto> {
    override fun isValid(
        value: ObjectMappingDto,
        hibernateConstraintValidatorContext: HibernateConstraintValidatorContext,
    ): Boolean {
        return value.valueMappingPerKey.isNotEmpty() ||
            value.valueCollectionMappingPerKey.isNotEmpty() ||
            value.objectMappingPerKey.isNotEmpty() ||
            value.objectCollectionMappingPerKey.isNotEmpty()
    }
}
